scipy/linalg/decomp_qr.py

```diff
@@ -84,28 +84,28 @@ def qr(a, overwrite_a=False, lwork=None, mode='full', pivoting=False,
 
     Examples
     --------
-    >>> from scipy import random, linalg, dot, diag, all, allclose
-    >>> a = random.randn(9, 6)
+    >>> from scipy import linalg
+    >>> a = np.random.randn(9, 6)
```

Comment:
Using random numbers without an explicit seed sometimes generates weird failures. 

That's a preexisting issue (and this docstring did not give us trouble before), so it does not block this PR.

Reply:
Our `plot_pre_code` is meant to ensure that we are okay:

https://github.com/scipy/scipy/blob/master/doc/source/conf.py#L379-L381

Reply:
Refguide-check disagrees, https://github.com/scipy/scipy/blob/6713b822b10ea7caadeb5ba78360ab80016accb1/tools/refguide_check.py#L669

I don't know who wins.

Reply:
On Sphinx the conf should win because of `plot_directive` makes use of it, in refguide check the `None` will win.

I guess this makes some sense as it makes our doc builds reproducible (by fixing random state during doc build), but at least to some extent makes it so that code will work when users copy-paste it (by not fixing random state in refguide check).

Reply:
Yup, thanks for the explanation Eric!